File: HackerRank Ransom Note.py
def ransom_note(magazine, note):
    freq = {}

    for word in magazine:
        freq[word] = freq.get(word, 0) + 1

    for word in note:
        if freq.get(word, 0) == 0:
            return "No"
        freq[word] -= 1

    return "Yes"

File: HackerRank Ransom Note.c
#include <stdio.h>
#include <string.h>

int main() {
    int m, n;
    scanf("%d %d", &m, &n);

    char magazine[1000][20], note[1000][20];

    for(int i = 0; i < m; i++)
        scanf("%s", magazine[i]);

    for(int i = 0; i < n; i++)
        scanf("%s", note[i]);

    int used[1000] = {0};

    for(int i = 0; i < n; i++) {
        int found = 0;

        for(int j = 0; j < m; j++) {
            if(!used[j] && strcmp(note[i], magazine[j]) == 0) {
                used[j] = 1;
                found = 1;
                break;
            }
        }

        if(!found) {
            printf("No");
            return 0;
        }
    }

    printf("Yes");
    return 0;
}
